sdks/java/io/elasticsearch-tests/elasticsearch-tests-2/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTest.java

```diff
@@ -101,6 +101,14 @@ public static void beforeClass() throws IOException {
     restClient = connectionConfiguration.createClient();
     elasticsearchIOTestCommon =
         new ElasticsearchIOTestCommon(connectionConfiguration, restClient, false);
+    while (restClient.performRequest("HEAD", "/").getStatusLine().getStatusCode() != 200) {
```

Comment:
Should there be an upper limit for the loop?

Reply:
I add max startup waiting time in the second commit